src/QueryPipeline/Pipe.cpp

```diff
@@ -170,8 +170,7 @@ Pipe::Pipe(ProcessorPtr source, OutputPort * output, OutputPort * totals, Output
 
 Pipe::Pipe(ProcessorPtr source)
 {
-    if (source->getOutputs().size() != 1)
```

Comment:
Why is it really unneeded?

Reply:
The `checkSource` always check does inputs is empty and output size is 1:

https://github.com/ClickHouse/ClickHouse/blob/b1c84aebc426419879aab0d3ebbaab1f3c277e09/src/QueryPipeline/Pipe.cpp#L23-L38

I think it should check inputs as well, even if we do not need to check inputs, the logic also seems wired, because it will check twice the size of outputs: outside `checkSource` function and insize `checkSource` function.